src/instructlab/configuration.py

```diff
@@ -376,27 +434,60 @@ class _generate(BaseModel):
 
     # model configuration
     model_config = ConfigDict(extra="ignore")
-
     # required fields
-    pipeline: Optional[str] = DEFAULTS.SDG_PIPELINE
-    model: StrictStr
-    taxonomy_path: StrictStr
-    taxonomy_base: StrictStr
-
+    pipeline: Optional[str] = Field(
+        default=DEFAULTS.SDG_PIPELINE,
+        description="Data generation pipeline to use. Available: simple, full, or a valid path to a directory of pipeline workflow YAML files. Note that 'full' requires a larger teacher model, Mixtral-8x7b.",
```

Comment:
nit: suggest to put `simple` and `full` in quotes to make it clear that these are literals.